### PR TITLE
Polish contract deletion flow

### PR DESCRIPTION
**Pull request by adam-maj**

### CODE REVIEW

**Comment by jnsdls** [comment]
#### 📝 Changed routes:


| [**/\[wallet\]**](https://thirdweb-jyx46ptk5-thirdweb.vercel.app/[wallet]) |
|:-:|
|<a href="https://thirdweb-jyx46ptk5-thirdweb.vercel.app/[wallet]">Screenshot of /[wallet]</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/edBt-CYKGF75cubm_6OEg5rmplAWJ14LOttSqxsDDDLqjYodcYWiXiXU7A16VYwVxwV-jz9w9GVQbuIkCxz1dA.png">(view full size)</a> |







Commit 9ee47056d9b42b5be2809f6c50c45988e8ede927 (https://thirdweb-jyx46ptk5-thirdweb.vercel.app).

**Comment by joaquim-verges** [comment]
any way we can optimistically update the list? feels so slow to update

**Comment by joaquim-verges** [comment]
also @adam-maj the network mismatch / deletion warning popover has the wrong background color

File: pages/[wallet]/index.tsx [hunk 1]
```diff
@@ -927,85 +931,108 @@ const RemoveContract: React.FC<IRemoveContract> = ({
...
+                <Text size="body.sm" fontStyle="italic">
+                  Your connected wallet does not support programatic switching.
+                  <br />
+                  Please manually switch the network in your wallet.
+                </Text>
+              )}
+            </Flex>
+          </PopoverBody>
+        ) : (
+          <PopoverBody>
```
Comment (joaquim-verges) [inline pages/[wallet]/index.tsx:995]: can we re-use `<MismatchButton>` for this?

File: pages/[wallet]/index.tsx [hunk 2]
```diff
@@ -927,85 +931,108 @@ const RemoveContract: React.FC<IRemoveContract> = ({
...
-        <PopoverBody>
-          <Flex direction="column" gap={4}>
-            <Heading size="label.lg">
-              <Flex gap={2} align="center">
-                <Icon boxSize={6} as={AiOutlineWarning} />
-                <span>Network Mismatch</span>
-              </Flex>
-            </Heading>
+        {chainId === targetChainId ? (
+          <PopoverBody>
```
Comment (joaquim-verges) [inline pages/[wallet]/index.tsx:955]: to get the right background here looks like you need to do what the mismatchButton does:

```
<Card
          maxW="sm"
          w="auto"
          as={PopoverContent}
          bg="backgroundCardHighlight"
          mx={6}
          boxShadow="0px 0px 2px 0px var(--popper-arrow-shadow-color)"
        >
        ```

**Comment by adam-maj** [comment]
> any way we can optimistically update the list? feels so slow to update. Might need to invalidate the contract list query

The contract list query is getting invalidated here - but it's going very slowly for some reason... I wonder if its just because the query itself is slow? Must be smt we can do about it cc @jnsdls

**Pull request merged**